src/App.js

```diff
@@ -7,30 +7,34 @@ const { default: CreateMuiTheme } = require('@material-ui/core/styles/createMuiT
 const { default: MuiThemeProvider } = require('@material-ui/styles/ThemeProvider');
 const { default: CssBaseline } = require('@material-ui/core/CssBaseline');
 const StrangeRouter = require('strange-router');
+const MiddleEnd = require('strange-middle-end');
 const { ConnectedRouter } = require('connected-react-router');
 const { default: ErrorBoundary } = require('react-error-boundary');
 const ErrorFallback = require('./components/ErrorFallback');
 const Routes = require('./routes');
 
-module.exports = ({ store, theme = CreateMuiTheme(), Router = ConnectedRouter, ...routerProps }) => {
+module.exports = ({ middleEnd, store, theme = CreateMuiTheme(), Router = ConnectedRouter, ...routerProps }) => {
 
     return (
         <Styled.ThemeProvider theme={theme}>
             <MuiThemeProvider theme={theme}>
                 <ErrorBoundary FallbackComponent={ErrorFallback}>
                     <CssBaseline />
-                    <ReactRedux.Provider store={store}>
-                        <Router {...routerProps}>
-                            <StrangeRouter.Routes routes={Routes} />
-                        </Router>
-                    </ReactRedux.Provider>
+                    <MiddleEnd.Provider middleEnd={middleEnd}>
+                        <ReactRedux.Provider store={store}>
+                            <Router {...routerProps}>
+                                <StrangeRouter.Routes routes={Routes} />
+                            </Router>
+                        </ReactRedux.Provider>
+                    </MiddleEnd.Provider>
                 </ErrorBoundary>
             </MuiThemeProvider>
         </Styled.ThemeProvider>
     );
 };
 
 module.exports.propTypes = {
+    middleEnd: T.object,
     store: T.object.isRequired,
```

Comment:
When we remove `store` above, we'll remove it from `propTypes` as well — do we want to make `middleEnd` `required`?

Also, do we need more specificity, like `middleEnd: T.shape({ store: T.object })`?

Reply:
I believe a middle-end is required for `App` to function, so 👍 to making `middleEnd` a required prop.  I don't see any harm in explicitly expressing that `store` is a required key too.